backend/zeno_backend/processing/chart.py

```diff
@@ -17,11 +17,74 @@
 from zeno_backend.classes.filter import FilterPredicateGroup, Join
 from zeno_backend.classes.metric import Metric
 from zeno_backend.classes.slice import Slice
-from zeno_backend.database.select import metrics, slices
+from zeno_backend.database.select import metrics, models, slices
 from zeno_backend.processing.filtering import table_filter
 from zeno_backend.processing.metrics.map import metric_map
 
 
+def get_selected_slices(chart_slices: list[int], project: str) -> list[Slice]:
+    """Get the slices of the chart from a list of slice ids.
+
+    Args:
+        chart_slices (list[int]): slice ids of the chart.
+        project (str): project the user is currently working with.
+
+    Returns:
+        list[Slice]: list of slices of the chart.
+    """
+    if -2 in chart_slices:  # -2 signals that all slices should be selected
+        return slices(project)
+    selected_slices = slices(project, chart_slices)
+    if -1 in chart_slices:  # -1 signals that the "all instances" slice should be added
+        index = chart_slices.index(-1)
+        selected_slices.insert(
+            index,
+            Slice(
+                id=-1,
+                slice_name="All instances",
+                filter_predicates=FilterPredicateGroup(
+                    predicates=[], join=Join.OMITTED
+                ),
+            ),
+        )
+    return selected_slices
+
+
+def get_selected_metrics(chart_metrics: list[int], project: str) -> list[Metric]:
+    """Get the metrics of the chart from a list of chart ids.
+
+    Args:
+        chart_metrics (list[int]): metric ids of the chart.
+        project (str): project the user is currenlty working with.
+
+    Returns:
+        list[Metric]: list of metrics of the chart.
+    """
+    if -2 in chart_metrics:  # -2 signals that all metrics should be selected
+        return metrics(project)
+    return list(
+        filter(
+            lambda metric: metric.id in chart_metrics,
+            metrics(project) + [Metric(id=-1, name="count", type="count", columns=[])],
+        )
+    )
+
+
+def get_selected_models(chart_models: list[str], project: str) -> list[str]:
+    """Get the models of the chart from a list of models.
+
+    Args:
+        chart_models (list[str]): the list of models of the chart.
+        project (str): project the user is currently working with.
+
+    Returns:
+        list[str]: list of models of the chart.
+    """
+    if "all_zeno_project_models" in chart_models:  # all models should be selected
```

Comment:
what if someone names a model this ahaha

Reply:
Yep that's a problem. The alternative would be adding another field but don't really like that either. Don't know what to do....

Reply:
does an empty string work?

Reply:
if not it's fine super edge case

Reply:
Very good idea!